tests/data/test_module.py

```diff
@@ -10,36 +11,83 @@
 
 @pytest.mark.skipif(GITHUB_ACTIONS, reason="only works locally")
 class TestModuleInstantiation:
-    def test_datamodule_initialization(self):
-        temp_data_args = ClimateDatasetArgs(variables=["2m_temperature"], split="train")
-        temp_highres_data_args = ClimateDatasetArgs(
-            variables=["2m_temperature"], split="train"
-        )
-        temp_task_args = DownscalingArgs(
-            temp_data_args,
-            temp_highres_data_args,
-            in_vars=["2m_temperature"],
-            out_vars=["2m_temperature"],
-        )
-        DataModuleArgs(temp_task_args, 2014, 2016, 2017)
-
-    def test_initialization(self):
-        temp_data_args = ERA5Args(
+    def test_map_initialization(self):
+        climate_dataset_args = ERA5Args(
             root_dir=os.path.join(DATA_PATH, "era5/5.625deg/"),
-            variables=["2m_temperature"],
-            years=range(2010, 2015),
+            variables=[
+                "2m_temperature",
+                "geopotential",
+                "temperature_250",
+                "orography",
+                "land_sea_mask",
+            ],
+            years=range(2009, 2015),
             split="train",
         )
-        temp_highres_data_args = ERA5Args(
+        high_res_climate_dataset_args = ERA5Args(
             root_dir=os.path.join(DATA_PATH, "era5/2.8125deg/"),
-            variables=["2m_temperature"],
-            years=range(2010, 2015),
+            variables=[
+                "geopotential",
+                "temperature_250",
+            ],
+            years=range(2009, 2015),
+            split="train",
+        )
+        stacked_climate_dataset_args = StackedClimateDatasetArgs(
+            [climate_dataset_args, high_res_climate_dataset_args]
+        )
+        downscaling_args = DownscalingArgs(
+            in_vars=[
+                "2m_temperature",
+                "geopotential",
+                "temperature_250",
+            ],
+            out_vars=[
+                "geopotential",
+                "temperature_250",
+            ],
+            constant_names=[
```

Comment:
From a user POV (not knowing how the code works under the hood), this design is not clear about the role of the constants. I think constants should be lumped in with `in_vars`.

Reply:
There are two choices. The first one is to separate `constants` with `variables`. Since `constants` are not really varying with time. The other choice to keep them together. I personally prefer the first over the other.

Currently we are somewhere in the middle as the ClimateDataset for instantiation purposes takes just `in_vars` and `out_vars` which includes both `variables` and `constants` whereas the `Task` explicitly takes `in_vars` and `constant_names`. To solve this discrepancy, I opened #67. 

The other benefit of the first choice is that, as we know some values are constants so there is no need to re-read the files in the case of `ShardDataset`. 

From the user's perspective, `constants` is just some extra information that is available to you other than `in_vars`. Although for hotfix when I create input, I just stack `constants` with the `in_vars` but from a long term perspective, we should give user the flexibility  to come up with a clever model that treats `constants` and `in_vars` differently. That is only possible if we keep this distinction.

Reply:
OK. Fair points.

Reply:
I ended up implementing #67. Now I have fully disentangled the `constants` and `in_vars` logic in the `ClimateDataset` class. `ClimateDatasetArgs` now also takes input the [`constants`](https://github.com/aditya-grover/climate-learn/blob/2f3db813b98ccb91b14709da7796734e5bdba99e/src/climate_learn/data/climate_dataset/args/climate_dataset_args.py#L18) at the initialization time.